libmamba/src/core/fetch.cpp

```diff
@@ -168,6 +205,11 @@ namespace mamba
         // - fixing how the progress bar works
         curl_easy_setopt(handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
 
+        if (std::getenv("MAMBA_SET_USEDPWD"))
+        {
+            curl_easy_setopt(handle, CURLOPT_USERPWD, ":test");
```

Comment:
@wolfv i'm pretty sure I rootcaused what was going on here.

I believe that with https://github.com/curl/curl/pull/6545/ (which was released with the release of libcurl where we noticed this starting to break for us), the fact that we never directly set the user/password on the curl handle but instead relied on it being passed through the URL led to some edge case in curl where the auth ends up being dropped from reusing the connection.

I'll follow up with an issue towards curl to see if we can get this sorted there.

Reply:
Filed https://github.com/curl/curl/issues/8449